CONTRIBUTING.md

```diff
@@ -0,0 +1,58 @@
+- Discussing the current state of the code
+- Submitting a fix
+- Proposing new features
+- Becoming a maintainer
+
+## We Develop with Github
+We use github to host code, to track issues and feature requests, as well as accept pull requests.
+
+## Setting up your Development Environment
+For information on how to set up your environment, go to the [Contributor's Guide](https://github.com/conjure-cp/conjure-oxide/wiki/Setting-up-your-development-environment)
+
+## We Use [Github Flow](https://guides.github.com/introduction/flow/index.html), So All Code Changes Happen Through Pull Requests
+Pull requests are the best way to propose changes to the codebase (we use [Github Flow](https://guides.github.com/introduction/flow/index.html)). We actively welcome your pull requests:
+
+1. Make a fork.
+2. Create a branch on your fork, do not develop on main.
+3. Create a pull request as soon as you want others to be able to see your progress, comment, and/or help. 
+4. Keep the PR in draft status until you think it's ready to be merged.
+5. Assign PR to reviewer(s) when it's ready to be merged.
+    - Only Oz (ozgurakgun) can merge PR's, so add him as a reviewer when you want your PR to be merged.
+7. We are happy to squash-merge, so feel free to push small commits to the PR. You can also tidy up the history (i.e. force-push) instead of a squash-merge if you choose to do that.
+
+## What We Didn't Do
+To prevent unknown unknowns, skim the documentation and [What We Didn't Do](https://github.com/conjure-cp/conjure-oxide/wiki/What-We-Didn%27t-Do).
+
+## Any contributions you make will be under the Mozilla Public License
+In short, when you submit code changes, your submissions will be understood to be under the same [Mozilla Public License](https://www.mozilla.org/en-US/MPL/2.0/) that covers the project. Feel free to contact the maintainers if that's a concern.
+
+## Report bugs using Github's [issues](https://github.com/conjure-cp/conjure-oxide/issues)
+We use GitHub issues to track public bugs. Report a bug by [opening a new issue](); it's that easy!
+
+## Write bug reports with detail, background, and sample code
+[This is an example](http://stackoverflow.com/q/12488905/180626) of a bug report I wrote, and I think it's not a bad model. Here's [another example from Craig Hockenberry](http://www.openradar.me/11905408), an app developer whom I greatly respect.
+
+**Great Bug Reports** tend to have:
+
+- A quick summary and/or background
+- Steps to reproduce
+  - Be specific!
+  - Give sample code if you can. [My stackoverflow question](http://stackoverflow.com/q/12488905/180626) includes sample code that *anyone* with a base R setup can run to reproduce what I was seeing
+- What you expected would happen
+- What actually happens
+- Notes (possibly including why you think this might be happening, or stuff you tried that didn't work)
+
+People *love* thorough bug reports. I'm not even kidding.
+
+## Coding Style
+Please write clear and well doccumented code.
```

Comment:
We can add here:

- Run `cargo fmt` in the project directory to automatically format code
- Use `cargo clippy` to lint the code and identify any common issues